lib/client.js

```diff
@@ -76,6 +78,34 @@ class Client {
       return Promise.resolve();
     }
     this.status = 1;
+    if (this._options.proxy.port) {
+      this.conn = SocksClient.createConnection(this._options).then(info => {
```

Comment:
is it possible to do this on a _level higher_? in BW actually, somewhere in `blue_modules/tls.js` ? 
otherwise, this turns into dependency hell, electrum-client depending on customs socks, which depend on custom node-ip